sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -235,7 +234,21 @@ abstract class QueryTest extends PlanTest {
   }
 }
 
-object QueryTest {
+object QueryTest extends Assertions {
+  /**
+   * Runs the plan and makes sure the answer matches the expected result.
+   *
+   * @param df the [[DataFrame]] to be executed
+   * @param expectedAnswer the expected result in a [[Seq]] of [[Row]]s.
+   * @param checkToRDD whether to verify deserialization to an RDD. This runs the query twice.
+   */
+  def checkAnswer(df: DataFrame, expectedAnswer: Seq[Row], checkToRDD: Boolean = true): Unit = {
```

Comment:
We can make it `private` if this isn't supposed to be called in the tests directly. companion class still can access.

Reply:
There are some suites needs this one: `ReduceNumShufflePartitionsSuite`, `SessionStateSuite` and `SQLQuerySuite`(`SQLQuerySuite` needs to set the parameter `checkToRDD` as false)
And I prefer not to have duplicated code in these 3 suites.